src/pages/profile/profile.ts

```diff
@@ -1,29 +1,56 @@
 import { Component, OnInit } from '@angular/core';
-import { NavController, AlertController } from 'ionic-angular';
+import { Pictures } from 'api/collections';
+import { DEFAULT_PICTURE_URL, Profile } from 'api/models';
+import { AlertController, NavController } from 'ionic-angular';
 import { MeteorObservable } from 'meteor-rxjs';
-import { Profile } from 'api/models/whatsapp-models';
-import { TabsPage } from "../tabs/tabs";
+import { PictureUploader } from '../../services/picture-uploader';
+import { TabsPage } from '../tabs/tabs';
 
 @Component({
   selector: 'profile',
   templateUrl: 'profile.html'
 })
-export class ProfileComponent implements OnInit {
+export class ProfilePage implements OnInit {
+  picture: string;
   profile: Profile;
 
   constructor(
-    public navCtrl: NavController,
-    public alertCtrl: AlertController
+    private alertCtrl: AlertController,
+    private navCtrl: NavController,
+    private pictureUploader: PictureUploader
   ) {}
 
   ngOnInit(): void {
     this.profile = Meteor.user().profile || {
-      name: '',
-      picture: '/ionicons/dist/svg/ios-contact.svg'
+      name: 'Whatsapp Newbie'
     };
+
+    MeteorObservable.subscribe('user').subscribe(() => {
+      const picture = Pictures.findOne(this.profile.pictureId) || {};
+      this.picture = picture.url || DEFAULT_PICTURE_URL;
+    });
+  }
+
+  selectProfilePicture(): void {
+    this.pictureUploader.select().then((file) => {
+      this.uploadProfilePicture(file);
+    });
+  }
+
+  uploadProfilePicture(file: File): void {
+    this.pictureUploader.upload(file).then((picture) => {
+      const oldPictureId = this.profile.pictureId;
+      if (oldPictureId) Meteor.call('removePicture', oldPictureId);
```

Comment:
I did this because I wanted to dispose old profile pictures, but on a second thought I'm not sure it's a good idea because usually all the data is kept so references won't be lost.